packages/back-end/src/api/visual-changesets/listVisualChangesets.ts

```diff
@@ -0,0 +1,30 @@
+import { ListVisualChangesetsResponse } from "../../../types/openapi";
+import {
+  findVisualChangesetsByOrganization,
+  toVisualChangesetApiInterface,
+} from "../../models/VisualChangesetModel";
+import { applyPagination, createApiRequestHandler } from "../../util/handler";
+import { listVisualChangesetsValidator } from "../../validators/openapi";
+
+export const listVisualChangesets = createApiRequestHandler(
+  listVisualChangesetsValidator
+)(
+  async (req): Promise<ListVisualChangesetsResponse> => {
+    const visualChangesets = await findVisualChangesetsByOrganization(
+      req.organization.id
+    );
+
+    // TODO: Move sorting/limiting to the database query for better performance
+    const { filtered, returnFields } = applyPagination(
+      visualChangesets.sort((a, b) => a.id.localeCompare(b.id)),
```

Comment:
Is this sorting alphabetically by a generated string ID? I didn't see any object validation regarding the pagination params. Not sure if we want to add that to the generators if they aren't there already. One potential regression I can imagine happening is if someone, for whatever reason, needs to add query params to an endpoint an existing object validation for query params isn't there, and they accidentally break pagination (by doing something like adding strict object validation for the query params).

Reply:
I removed this .. See my comment